src/lib/FileManager.js

```diff
@@ -105,13 +105,17 @@ export default class FileManager {
   }
 
   async downloadFileFromDescriptor(fileDescriptor) {
-    var retries = 3;
-    var integration = this.integrationManager.integrationForFileDescriptor(fileDescriptor);
-    while (retries-- > 0 && !integration) {
-      setTimeout(function() {
-        integration = this.integrationManager.integrationForFileDescriptor(fileDescriptor);
-      }, 1000);
+    function getIntegration(retries) {
+      var integrationTry = this.integrationManager.integrationForFileDescriptor(fileDescriptor);
+      if (integrationTry || retries <= 0) {
+        return integrationTry;
+      } else {
+        return setTimeout(getIntegration(--retries), 1000);
+      }
     }
+    var numRetries = 3;
+    var integration = getIntegration(numRetries);
```

Comment:
This function still wouldn't wait 3 * 1000ms to return. It would return immediately actually.

The only good way to block code execution while waiting some number of milliseconds is to wrap setTimeout in an async function: https://github.com/standardnotes/snjs/blob/004/lib/utils.ts#L443

But overall I feel like this PR hacks at the ultimately problem without understanding what it is, and in that manner doesn't add much value, even if it luckily fixes the problem. We really should understand what the root cause is, and solve that problem. Because as-is, it's only by some miracle that waiting 3 seconds causes it to work. Sounds like a race condition, where could is called earlier than it should be, or called at the same time when instead it should be called serially.

Reply:
That is true. More investigation is required. Have you ever faced this issue?

Reply:
I haven't personally. Have you? We want to make sure that in the code, no images are attempting to load before EditorKit has received integrations. Or, we should attempt reload of images after a new integration is received.